src/main/java/com/synopsys/integration/detect/configuration/DetectConfigurationFactory.java

```diff
@@ -169,9 +171,23 @@ public BlackDuckSignatureScannerOptions createBlackDuckSignatureScannerOptions()
 
         final String userProvidedScannerInstallUrl = detectConfiguration.getProperty(DetectProperty.DETECT_BLACKDUCK_SIGNATURE_SCANNER_HOST_URL, PropertyAuthority.None);
 
+        String snippetMatchingString = detectConfiguration.getProperty(DetectProperty.DETECT_BLACKDUCK_SIGNATURE_SCANNER_SNIPPET_MATCHING, PropertyAuthority.None).trim().toUpperCase();
```

Comment:
Sigh. This is fine but extremely unfortunate that the library doesn't do a better job of supporting this. I think what we'd want is something that validates the scanner configuration in the library so that detect could make use of that. Can you create a ticket to build that feature?

Reply:
I created a PR for doing the validation check on upload source and snippet matching.
https://github.com/blackducksoftware/blackduck-common/pull/264